pkg/controllermanager/controller/managedseedset/managedseedset_reconciler.go

```diff
@@ -0,0 +1,160 @@
+		}
+		r.logger.Errorf("Could not get ManagedSeedSet %s from store: %+v", request.NamespacedName, err)
+		return reconcile.Result{}, err
+	}
+
+	if set.DeletionTimestamp != nil {
+		return r.delete(ctx, set)
+	}
+	return r.reconcile(ctx, set)
+}
+
+func (r *reconciler) reconcile(ctx context.Context, set *seedmanagementv1alpha1.ManagedSeedSet) (result reconcile.Result, err error) {
+
+	// Ensure gardener finalizer
+	r.getLogger(set).Debug("Ensuring finalizer")
+	if err := controllerutils.PatchAddFinalizers(ctx, r.gardenClient.Client(), set, gardencorev1beta1.GardenerName); err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not ensure gardener finalizer: %w", err)
+	}
+
+	var status *seedmanagementv1alpha1.ManagedSeedSetStatus
+	defer func() {
+		// Update status, on failure return the update error unless there is another error
+		if updateErr := r.updateStatus(ctx, set, status); updateErr != nil && err == nil {
+			err = fmt.Errorf("could not update status: %w", updateErr)
+		}
+	}()
+
+	// Reconcile creation or update
+	r.getLogger(set).Debug("Reconciling creation or update")
+	if status, err = r.actuator.Reconcile(ctx, set); err != nil {
+		r.errorEventf(set, "Could not reconcile creation or update: %+v", err)
+		return reconcile.Result{}, fmt.Errorf("could not reconcile ManagedSeedSet %s creation or update: %w", kutil.ObjectName(set), err)
+	}
+	r.getLogger(set).Debug("Creation or update reconciled")
+
+	// Return success result
+	return reconcile.Result{RequeueAfter: r.cfg.SyncPeriod.Duration}, nil
+}
+
+func (r *reconciler) delete(ctx context.Context, set *seedmanagementv1alpha1.ManagedSeedSet) (result reconcile.Result, err error) {
+	// Check gardener finalizer
+	if !controllerutil.ContainsFinalizer(set, gardencorev1beta1.GardenerName) {
+		r.getLogger(set).Debug("Skipping as it does not have a finalizer")
+		return reconcile.Result{}, nil
+	}
+
+	var status *seedmanagementv1alpha1.ManagedSeedSetStatus
+	defer func() {
+		// Update status, on failure return the update error unless there is another error
+		if updateErr := r.updateStatus(ctx, set, status); updateErr != nil && err == nil {
+			err = fmt.Errorf("could not update status: %w", updateErr)
+		}
+	}()
+
+	// Reconcile deletion
+	r.getLogger(set).Debug("Reconciling deletion")
+	if status, err = r.actuator.Reconcile(ctx, set); err != nil {
+		r.errorEventf(set, "Could not reconcile deletion: %+v", err)
+		return reconcile.Result{}, fmt.Errorf("could not reconcile ManagedSeedSet %s deletion: %w", kutil.ObjectName(set), err)
+	}
+	r.getLogger(set).Debug("Deletion reconciled")
+
+	// If there is no pending replica, remove gardener finalizer
+	if status.PendingReplica == nil {
+		r.getLogger(set).Debug("Removing finalizer")
+		if err := controllerutils.PatchRemoveFinalizers(ctx, r.gardenClient.Client(), set, gardencorev1beta1.GardenerName); err != nil {
+			return reconcile.Result{}, fmt.Errorf("could not remove gardener finalizer: %w", err)
+		}
```

Comment:
We should return without requeueing after successfully removing the finalizer
```suggestion
		}
		return reconcile.Result{}, nil
```

Reply:
OK, yes, good point.